core/src/main/java/org/apache/calcite/plan/RelTraitDef.java

```diff
@@ -147,6 +147,7 @@ public abstract boolean canConvert(
    * @param fromRel   the RelNode to convert from (with fromTrait)
    * @return true if fromTrait can be converted to toTrait
    */
+  @Deprecated // to be removed before 1.24
```

Comment:
This can be moved to the JavaDoc?

Reply:
you mean L148?

Reply:
I think in the JavaDoc of a deprecated method, there should be a line marked as @deprecated, and provide some instructions for the client (e.g. please use another method instead, or the method will no longer be supported)

So I think line 150 should be moved to the JavaDoc?

Reply:
Sorry, I noticed this way is common in the Calcite code base. So this is not a problem.